### PR TITLE
Improve worker implementation

### PR DESCRIPTION
**Pull request by martinRenou**
cc @SylvainCorlay @maartenbreddels 

I've put the workers implementation in a `SafeJSKernel` class (I'm open to better names if you've got ideas). 

This implementation is faster than the old one, we only create one `Worker` per `Renderer` instead of one `Worker` per rendering function call.

**NOTE:** I also started to really use TypeScript here

### CODE REVIEW

File: js/src/renderer.ts [hunk 1]
```diff
@@ -1,33 +1,140 @@
...
+        this.worker.onmessage = (evt: any) => {
+            // Resolve the right Promise with the return value
+            this.requests[evt.data.id].resolve(evt.data.result);
+            delete this.requests[evt.data.id];
+        };
+
+        this.worker.onerror = (evt: any) => {
+            // Reject all the pending promises, terminate the worker and start again
+            forEach(this.requests, (request, id) => {
+                request.reject(new Error(evt.message));
```
Comment (martinRenou) [inline js/src/renderer.ts]: I'm not sure the `new Error` is needed here.

File: js/src/renderer.ts [hunk 2]
```diff
@@ -1,33 +1,140 @@
...
-        // We add Handsontable manually as extra argument to put it in the scope
+        };
+    }
+
+    initialize(attributes: any, options: any) {
+        super.initialize(attributes, options);
+
+        this.kernel = new SafeJSKernel();
+
         var that = this;
```
Comment (martinRenou) [inline js/src/renderer.ts]: We should get rid of this

File: js/src/renderer.ts [hunk 3]
```diff
@@ -1,33 +1,140 @@
...
-        (Handsontable.renderers as any).registerRenderer(this.get('name'), this.fn);
+
+        (Handsontable.renderers as any).registerRenderer(this.get('name'), this.rendering_function);
     }
-});
 
-export {
-    RendererModel 
+    kernel: SafeJSKernel;
+    rendering_function: any;
```
Comment (martinRenou) [inline js/src/renderer.ts]: Missing type

**Comment by SylvainCorlay** [comment]
Thanks! I did not notice the PR until now. I will review by Monday!

**Pull request merged**